js/web/lib/wasm/jsep/backend-webgpu.ts

```diff
@@ -701,32 +701,31 @@ export class WebGpuBackend {
     }
   }
 
-  captureBegin(sessionHandle: number): void {
-    LOG_DEBUG('info', () => `captureBegin ${sessionHandle}`);
-    this.currentSessionId = sessionHandle;
-    let sessionCommandList = this.capturedCommandList.get(sessionHandle);
-    let sessionPendingKernels = this.capturedPendingKernels.get(sessionHandle);
+  captureBegin(): void {
+    LOG_DEBUG('info', () => 'captureBegin');
+    let sessionCommandList = this.capturedCommandList.get(this.currentSessionId!);
+    let sessionPendingKernels = this.capturedPendingKernels.get(this.currentSessionId!);
     if (!sessionCommandList) {
       sessionCommandList = [];
-      this.capturedCommandList.set(sessionHandle, sessionCommandList);
+      this.capturedCommandList.set(this.currentSessionId!, sessionCommandList);
       sessionPendingKernels = [];
-      this.capturedPendingKernels.set(sessionHandle, sessionPendingKernels);
+      this.capturedPendingKernels.set(this.currentSessionId!, sessionPendingKernels);
     }
-    this.status = StatusType.capture;
+    // flush the left commands before we change the status.
+    this.flush();
+    this.sessionStatus = 'capturing';
   }
-  captureEnd(sessionHandle: number): void {
-    LOG_DEBUG('info', () => `captureEnd ${sessionHandle}`);
+  captureEnd(): void {
+    LOG_DEBUG('info', () => 'captureEnd');
     // flush the left commands before we change the status.
     this.flush();
-    this.currentSessionId = null;
-    this.status = StatusType.default;
+    this.sessionStatus = 'default';
   }
-  replay(sessionHandle: number): void {
-    LOG_DEBUG('info', () => `replay ${sessionHandle}`);
-    this.currentSessionId = sessionHandle;
-    this.status = StatusType.replay;
-    const sessionCommandList = this.capturedCommandList.get(sessionHandle);
-    const sessionPendingKernels = this.capturedPendingKernels.get(sessionHandle);
+  replay(): void {
+    LOG_DEBUG('info', () => 'replay');
```

Comment:
if it's simple string, use string instead of function as 2nd parameter

The 2nd parameter is designed to be either string or a function that returns string. sometimes concating the string is heavy so make it a function avoid expression evaluation if the log level/debug flag has an early exit. But if it's a simple string there is no need to pass a function

Reply:
Done, thanks for your detail explanation.